pyresample/test/test_gradient.py

```diff
@@ -261,6 +263,313 @@ def test_resample_swath_to_area_3d(self):
             assert np.allclose(arr[np.isfinite(arr)], float(i + 1))
 
 
+class TestRBGradientSearchResamplerArea2Area:
+    """Test RBGradientSearchResampler for the Area to Area case."""
+
+    def setup(self):
+        """Set up the test case."""
+        from pyresample.gradient import ResampleBlocksGradientSearchResampler
+        self.src_area = AreaDefinition('src', 'src area', None,
+                                       {'ellps': 'WGS84', 'h': '35785831', 'proj': 'geos'},
+                                       100, 100,
+                                       (5550000.0, 5550000.0, -5550000.0, -5550000.0))
+
+        self.dst_area = AreaDefinition('euro40', 'euro40', None,
+                                       {'proj': 'stere', 'lon_0': 14.0,
+                                        'lat_0': 90.0, 'lat_ts': 60.0,
+                                        'ellps': 'bessel'},
+                                       102, 102,
+                                       (-2717181.7304994687, -5571048.14031214,
+                                        1378818.2695005313, -1475048.1403121399))
+
+        self.resampler = ResampleBlocksGradientSearchResampler(self.src_area, self.dst_area)
+
+    def test_precompute_generates_indices(self):
+        self.resampler.precompute()
+        assert self.resampler.indices_xy.shape == (2, ) + self.dst_area.shape
+
+    def test_resampler_accepts_only_dataarrays_if_not_2d(self):
+        data = da.ones(self.src_area.shape + (1,), dtype=np.float64, chunks=40)
+        self.resampler.precompute()
+        with pytest.raises(TypeError):
+            self.resampler.compute(data, method='bilinear').compute(scheduler='single-threaded')
+
+    def test_resampler_returns_the_right_shape(self):
+        data = xr.DataArray(da.ones((3, ) + self.src_area.shape,
+                                    dtype=np.float64) *
+                            np.array([1, 2, 3])[:, np.newaxis, np.newaxis],
+                            dims=['bands', 'y', 'x'])
+        self.resampler.precompute()
+        res = self.resampler.compute(
+            data, method='bilinear')
+        assert res.shape == (3, ) + self.dst_area.shape
+
+    def test_resampler_returns_a_dataarray(self):
+        data = xr.DataArray(da.ones(self.src_area.shape, dtype=np.float64),
+                            dims=['y', 'x'])
```

Comment:
This fake data creation seems to be repeated a lot, could it be put in a helper function? Or could all of these methods be rewritten with a combination of `pytest.mark.parametrize` parameters (ex. shape of input, attrs).

Reply:
parametrization is difficult as almost all test do test different things. I refactored the data creation using small helper functions though.